modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/filter/ComplexFilterSplitter.java

```diff
@@ -121,7 +138,116 @@ public Object visit(Id filter, Object notUsed) {
         
         return null;
     }
-    
+
+    @Override
+    public Object visit(Function expression, Object notUsed) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        Object data = super.visit(expression, notUsed);
+        // encoding of functions with nested attributes as  arguments is not supported
+        if (nestedAttributes > 0 && preStack.size() == i + 1) {
+            Object o = preStack.pop();
+            postStack.push(o);
+        }
+        return data;
+    }
+
+    @Override
+    protected void visitBinarySpatialOperator(BinarySpatialOperator filter) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        super.visitBinarySpatialOperator(filter);
+        // encoding of binary spatial operators operating on nested attributes is not supported
+        if (nestedAttributes > 0 && preStack.size() == i + 1) {
+            Object o = preStack.pop();
+            postStack.push(o);
+        }
+    }
+
+    @Override
+    protected Object visit(BinaryTemporalOperator filter, Object data) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        Object ret = super.visit(filter, data);
+        // encoding of temporal operators involving nested attributes is not supported
+        if (nestedAttributes > 0 && preStack.size() == i + 1) {
+            Object o = preStack.pop();
+            postStack.push(o);
+        }
+        return ret;
+    }
+
+    @Override
+    protected void visitMathExpression(BinaryExpression expression) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        super.visitMathExpression(expression);
+        // encoding of math expressions involving nested attributes is not supported
+        if (nestedAttributes > 0 && preStack.size() == i + 1) {
+            Object o = preStack.pop();
+            postStack.push(o);
+        }
+    }
+
+    @Override
+    public Object visit(BBOX filter, Object notUsed) {
+        nestedAttributes = 0;
+        int i = preStack.size();
+        if (filter.getExpression1() instanceof PropertyName) {
+            PropertyName bboxProperty = (PropertyName)filter.getExpression1();
+            if (!bboxProperty.getPropertyName().isEmpty()) {
+                Object ret = this.visit(bboxProperty, notUsed);
+                if (preStack.size() == i+1) {
+                    preStack.pop();
+                }
+                // encoding bbox on nested geometry is not supported
+                if (nestedAttributes > 0) {
+                    postStack.push(filter);
```

Comment:
OK.. so I get it. This is for the case when it's not supported. So nestedAttributes is meant to be calculated before this then?